### ISSUE TITLE
Error: The field is registered multiple times

### ISSUE THREAD

**Issue by cknitt** [DZakh/rescript-struct#63]
Hi @DZakh! I just tested rescript-struct 5 and really love the API improvements.

However, I ran into the following problem:

```rescript
type appVersionSpec = {
  current: string,
  minimum: string,
}

let appVersionSpecStruct = S.object(s => {
  current: s.field("current", S.string),
  minimum: s.field("minimum", S.string),
})

type appVersions = {ios: appVersionSpec, android: appVersionSpec}

let appVersionsStruct = S.object(s => {
  ios: s.field("ios", appVersionSpecStruct),
  android: s.field("android", appVersionSpecStruct),
})

let appVersions = {
  ios: {current: "1.1", minimum: "1.0"},
  android: {current: "1.2", minimum: "1.1"},
}

let _ = appVersions->S.serializeOrRaiseWith(appVersionsStruct)
```

gives me the exception:

```
RescriptStructError: Failed serializing at ["android"]. Reason: The field "current" is registered multiple times. If you want to duplicate the field, use S.transform instead
```

**Comment by DZakh** [comment]
Thanks for reporting the issue. I'll fix it today.

**Comment by DZakh** [comment]
I've published `v5.0.1` with a fix.

**Comment by cknitt** [comment]
Thanks for the quick fix! 👍
